plugin/src/main/cpp/register_types.cpp

```diff
@@ -81,74 +81,180 @@
-			OpenXRFbSpatialEntityExtensionWrapper::get_singleton()->register_extension_wrapper();
-
-			ClassDB::register_class<OpenXRFbSpatialEntitySharingExtensionWrapper>();
-			OpenXRFbSpatialEntitySharingExtensionWrapper::get_singleton()->register_extension_wrapper();
-
-			ClassDB::register_class<OpenXRFbSpatialEntityStorageExtensionWrapper>();
-			OpenXRFbSpatialEntityStorageExtensionWrapper::get_singleton()->register_extension_wrapper();
-
-			ClassDB::register_class<OpenXRFbSpatialEntityStorageBatchExtensionWrapper>();
-			OpenXRFbSpatialEntityStorageBatchExtensionWrapper::get_singleton()->register_extension_wrapper();
 
-			ClassDB::register_class<OpenXRFbSpatialEntityQueryExtensionWrapper>();
-			OpenXRFbSpatialEntityQueryExtensionWrapper::get_singleton()->register_extension_wrapper();
-
-			ClassDB::register_class<OpenXRFbSpatialEntityContainerExtensionWrapper>();
-			OpenXRFbSpatialEntityContainerExtensionWrapper::get_singleton()->register_extension_wrapper();
-
-			ClassDB::register_class<OpenXRFbSpatialEntityUserExtensionWrapper>();
-			OpenXRFbSpatialEntityUserExtensionWrapper::get_singleton()->register_extension_wrapper();
-
-			ClassDB::register_class<OpenXRMetaSpatialEntityMeshExtensionWrapper>();
-			OpenXRMetaSpatialEntityMeshExtensionWrapper::get_singleton()->register_extension_wrapper();
+static void define_global_bool(ProjectSettings *project_settings, const String &p_name, bool p_default_value) {
+	if (!project_settings->has_setting(p_name)) {
+		project_settings->set_setting(p_name, p_default_value);
+	}
 
-			ClassDB::register_class<OpenXRFbSceneExtensionWrapper>();
-			OpenXRFbSceneExtensionWrapper::get_singleton()->register_extension_wrapper();
+	project_settings->set_initial_value(p_name, true);
+	project_settings->set_as_basic(p_name, false);
+	Dictionary property_info;
+	property_info["name"] = p_name;
+	property_info["type"] = Variant::Type::BOOL;
+	property_info["hint"] = PROPERTY_HINT_NONE;
+	project_settings->add_property_info(property_info);	
+}
 
-			ClassDB::register_class<OpenXRFbFaceTrackingExtensionWrapper>();
-			OpenXRFbFaceTrackingExtensionWrapper::get_singleton()->register_extension_wrapper();
+static bool global_bool_get(const String &p_name) {
+	ProjectSettings *project_settings = ProjectSettings::get_singleton();
+	if (project_settings == nullptr) {
+		return false;
+	}
 
-			ClassDB::register_class<OpenXRFbBodyTrackingExtensionWrapper>();
-			OpenXRFbBodyTrackingExtensionWrapper::get_singleton()->register_extension_wrapper();
+	return project_settings->get_setting_with_override(p_name);
+}
 
-			ClassDB::register_class<OpenXRFbHandTrackingMeshExtensionWrapper>();
-			OpenXRFbHandTrackingMeshExtensionWrapper::get_singleton()->register_extension_wrapper();
 
-			ClassDB::register_class<OpenXRFbHandTrackingAimExtensionWrapper>();
-			OpenXRFbHandTrackingAimExtensionWrapper::get_singleton()->register_extension_wrapper();
+void add_plugin_core_settings() {
```

Comment:
I think we probably want all these internal functions to be `static`?

```suggestion
void static add_plugin_core_settings() {
```